File: android/app/src/main/java/com/example/messenger/data/local/db/entities/UserEntity.kt
package com.example.messenger.data.local.db.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.messenger.data.network.models.chat.list.User
import com.example.messenger.data.network.models.user.UserResponse

@Entity(tableName = "users")
data class UserEntity(
    @PrimaryKey
    val id:String,
    val created_at: String,
    val email: String,
    val first_name: String,
    val last_name: String,
    val nick_name: String,
    val phone: String,
    val updated_at: String,
    val avatar:String?,
    val isOnline:Boolean
)

fun User.toEntity() = UserEntity(
    id = id,
    created_at = created_at,
    email = email,
    first_name = first_name,
    last_name = last_name,
    nick_name = nick_name,
    phone = phone,
    updated_at = updated_at,
    avatar = avatar?.path,
    isOnline = isOnline
)

fun UserResponse.toEntity() = UserEntity(
    id = id,
    created_at = created_at,
    email = email,
    first_name = first_name,
    last_name = last_name,
    nick_name = nick_name,
    phone = phone,
    updated_at = updated_at,
    avatar = avatar?.path,
    isOnline = false
)

File: backend/src/typeorm/entities/user.entity.ts
import {
  Column,
  Entity,
  JoinColumn, JoinTable,
  ManyToMany,
  OneToOne,
  PrimaryGeneratedColumn
} from "typeorm";
import { BaseEntity } from './base.entity';
import { ChatEntity } from './chat.entity';
import { LocalFileEntity } from './local-file.entity';
import { MessageEntity } from './message.entity';

@Entity('users')
export class User extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  first_name: string;

  @Column()
  last_name: string;

  @Column()
  nick_name: string;

  @Column({ nullable: false, unique: true })
  phone: string;

  @Column({ default: false })
  isOnline: boolean;

  @Column({ nullable: false, unique: true })
  email: string;

  @JoinColumn({ name: 'avatarId' })
  @OneToOne(() => LocalFileEntity, { nullable: true })
  avatar?: LocalFileEntity;

  @Column({ nullable: true, select: false })
  public avatarId?: string;

  @Column({ nullable: false, select: false })
  passwordHash: string;

  @Column({ nullable: true, select: false })
  refreshToken: string;

  @ManyToMany(() => ChatEntity, (chatEntity) => chatEntity.users)
  @JoinTable()
  chats: ChatEntity[];

  @ManyToMany(() => MessageEntity, (messageEntity) => messageEntity.author)
  messages: MessageEntity[];
}
